qiskit_experiments/analysis/data_processing.py

```diff
@@ -177,7 +177,15 @@ def level2_probability(data: Dict[str, any], outcome: str) -> Tuple[float]:
         mean probability is :math:`p = K / N` and the variance is
         :math:`\\sigma^2 = p (1-p) / N`.
     """
+    # TODO fix sigma definition
+    # When the count is 100% zero (i.e. simulator), this yields sigma=0.
+    # This crashes scipy fitter when it calculates covariance matrix (zero-div error).
+
     counts = data["counts"]
+
+    if outcome is None:
+        outcome = "1" * len(list(counts.keys())[0])
```

Comment:
I think for this function it would be better to change how it is accessed as a data processor by making a function return like:

```python
def probability(outcome: str:) -> Callable:
    """Return data processing function"""
    def data_processor(data):
        return level2_probability(data, outcome)
    
    return data_processor
```
And then for the all 0 and all 1 special cases we can make new functions that work with variable qubit numbers along the lines of

```python
def probability0(data):
    counts = data["counts"]
    outcome = "1" * len(list(counts.keys())[0])
    return level2_probability(data, outcome)
```
and similarly for `probability0`.

Also these functions could be converted to use the `DataProcessor` class rather than just being function calls.